pkg/metadata/host/host.go

```diff
@@ -249,21 +250,47 @@ func getMeta() *Meta {
 	return m
 }
 
-func getContainerMeta() map[string]string {
+func getContainerMeta(timeout time.Duration) map[string]string {
+	wg := sync.WaitGroup{}
 	containerMeta := make(map[string]string)
+	// protecting the above map from concurrent access
+	mutex := &sync.Mutex{}
 
 	for provider, getMeta := range container.DefaultCatalog {
-		meta, err := getMeta()
-		if err != nil {
-			log.Debugf("Unable to get %s metadata: %s", provider, err)
-			continue
-		}
-		for k, v := range meta {
-			containerMeta[k] = v
+		wg.Add(1)
```

Comment:
The host metadata are fetch once every 4 hours, do we need to add the complexity of fetching container metadata in parallel ? How many providers are we expecting at the same time ? Could we not just call them sequentially with a 1s timeout ?

Reply:
we'll have at least 6 providers for now, and all of them will be fetching from APIs

If everything times out that would be 6 seconds, I thought it was wiser to fetch them in parallel to guarantee the 1 second for every provider.

Reply:
Could all 6 of them be used at the same time ? I think 6 seconds every 4hours is fine (and it doesn't block a worker or something).

If we decide to keep the goroutine we should log an error line in the timeout case.

Reply:
I agree that we don't care about the total time when it's called every 4 hours. But the one case we do care about is at agent startup: we collect host metadata then (see https://github.com/DataDog/datadog-agent/blob/6.2.1/pkg/metadata/scheduler.go#L83-L91) and want that payload to be sent as early as possible to avoid sending too many metrics/events/service checks before that metadata payload reaches the backend.

Reply:
ok, let's keep the goroutine then.